sunpy/util/scraper.py

```diff
@@ -343,22 +365,74 @@ def _extract_files_meta(self, timerange, extractor=None):
             Time interval where to find the directories for a given pattern.
         extractor: `str`
             Pattern to extract metadata by parsing the URL.
+        matcher: `dict`
+            Dictionary to check if extracted metadata is valid.
 
         Returns
         -------
-        metalist: `list` of `dict`
+        `list` of `dict`
             List of metadata info for all URLs.
         """
+        self.extractor = extractor
         urls = self.filelist(timerange)
-        if extractor is None:
-            return [{'url': url} for url in urls]
         metalist = []
         for url in urls:
             metadict = parse(extractor, url)
             if metadict is not None:
+                append = True
                 metadict = metadict.named
-            else:
-                metadict = {}
-            metadict['url'] = url
-            metalist.append(metadict)
+                metadict['url'] = url
+                if matcher:
+                    for k in metadict:
+                        if k in matcher and str(metadict[k]) not in matcher[k]:
+                            append = False
+                            break
+                if append:
+                    metalist.append(metadict)
         return metalist
+
+
+def get_timerange_from_exdict(exdict):
+    """
+    Function to get URL's timerange using extracted metadata.
```

Comment:
This docstring needs a bit more of explanation. It took me a bit to really understand what it does... and I'm not 100% sure. Does it provide a TimeRange based on the smaller input in `exdict`?

Reply:
Yes, it uses the day, month year, etc values which are in exdict extracted from url using the pattern to find start and end times.
I have added that in docstrings.